### PR TITLE
Chore/mini grid design fixes

### LINKED ISSUE #4274: Small design tweaks and minor fixes for the Grid panel layout

### PR DESCRIPTION
**Pull request by balazsbajorics**
Fixes #4274 

Mini design fixes for the Grid Panels system

**Commit Details:**
- Panel reorder: Re-enabled the logic that only shows a dropline if dropping the panel there would result in a layout change
- Panel reorder: the area above the canvas is a drop target now so you can easily reorder panels to the other side of the editor
- The code editor has a somewhat nicer title bar made by Malte last friday
- The code editor title bar's minimize and maximize buttons are wired in:
  - Minimize moves it below the Navigator
  - Maximize move it to be the leftmost panel
- A fix of an accidental mutation in an update function
- The panel layout's padding around the window border is significantly reduced to 8px
- Hiding that weird little square thumbnail that would travel with the reordered panels :)

### CODE REVIEW

File: editor/src/components/canvas/grid-panels-state.tsx [hunk 1]
```diff
@@ -189,23 +192,21 @@ export function updateLayout(
...
       const working = [...layout]
 
       // insert
-      working[update.columnIndex].panels = insert(
-        update.indexInColumn + 1,
-        panelToInsert,
-        working[update.columnIndex].panels,
-      )
+      working[update.columnIndex] = {
+        ...working[update.columnIndex],
```
Comment (seanparsons) [inline editor/src/components/canvas/grid-panels-state.tsx:207]: Extracting out the existing value into a variable would probably make this a little cleaner.

**Review by bkrmendy — APPROVED** [review]

File: editor/src/components/canvas/grid-panels-state.tsx [hunk 2]
```diff
@@ -189,23 +192,21 @@ export function updateLayout(
       const working = [...layout]
 
       // insert
-      working[update.columnIndex].panels = insert(
-        update.indexInColumn,
-        panelToInsert,
-        working[update.columnIndex].panels,
-      )
+      working[update.columnIndex] = {
+        ...working[update.columnIndex],
```
Comment (seanparsons) [inline editor/src/components/canvas/grid-panels-state.tsx:196]: Extracting out the existing value into a variable would probably make this a little cleaner.

File: editor/src/components/titlebar/title-bar.tsx [hunk 3]
```diff
@@ -334,6 +340,57 @@ export const TitleBarEmpty = React.memo((props: { panelData: StoredPanel }) => {
...
+        onClick={onMaximize}
+        style={{
+          width: 8,
+          height: 8,
+          borderRadius: 8,
+          backgroundColor: '#61C454',
+        }}
+      />
+      <span style={{ marginLeft: 8 }}>Code </span>
+      {/* <SquareButton highlight>+</SquareButton> */}
```
Comment (seanparsons) [inline editor/src/components/titlebar/title-bar.tsx:389]: To the bin?

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/titlebar/title-bar.tsx (hunk 3 above)
Comment (balazsbajorics) [reply editor/src/components/titlebar/title-bar.tsx:389]:
```suggestion
      {/* <SquareButton highlight>+</SquareButton> */}
```

**Review by gbalint — APPROVED** [review]

**Review by enidemi — APPROVED** [review]

**Pull request merged**